connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/util/IntegrationTestUtils.scala

```diff
@@ -49,6 +51,12 @@ object IntegrationTestUtils {
   // scalastyle:on println
   private[connect] def debug(error: Throwable): Unit = if (isDebug) error.printStackTrace()
 
+  private[sql] lazy val sparkHiveJarAvailable: Boolean = {
+    val filePath = s"$sparkHome/assembly/target/$scalaDir/jars/" +
+      s"spark-hive_$scalaVersion-${org.apache.spark.SPARK_VERSION}.jar"
```

Comment:
Does this cover SBT build too? I think it would be easier if you just try to get a class (e.g., `org.apache.hadoop.hive.conf.HiveConf`) and see if it fails.

Reply:
Yes, cover SBT to

Reply:
like `Utils.classIsLoadable("org.apache.hadoop.hive.conf.HiveConf")` ?

Reply:
> Does this cover SBT build too? I think it would be easier if you just try to get a class (e.g., `org.apache.hadoop.hive.conf.HiveConf`) and see if it fails.

Let me think about it

Reply:
@HyukjinKwon It is not ok to try to get the hive-related class on the client side(like `Utils.classIsLoadable("org.apache.hadoop.hive.conf.HiveConf")`),  the result should always be false.

`SimpleSparkConnectService` can find `spark-hive_**.jar` in classpath because of the following code:

https://github.com/apache/spark/blob/93334e295483a0ba66e22d8398512ad970a3ea80/bin/spark-class#L39-L51

Reply:
I manually verified the SBT. As I said above, if the code is changed to 

```
  private[sql] lazy val sparkHiveJarAvailable: Boolean = {
    Utils.classIsLoadable("org.apache.hadoop.hive.conf.HiveConf")
  }
```

Both 

`build/sbt clean "connect-client-jvm/test"`

and 

`build/sbt clean "connect-client-jvm/test" -Phive`

 will submit `SimpleSparkConnectService ` with `spark.sql.catalogImplementation=in-memory`

Reply:
🆗